app/views/admin/AdministerUserModal.coffee

```diff
@@ -389,6 +394,46 @@ module.exports = class AdministerUserModal extends ModelModal
           @renderSelectors('#'+prepaidId)
         return
 
+  onClickEditProductInfoButton: (e) ->
+    productId=@$(e.target).data('product-id')
+    @productTableState[productId] = 'editMode'
+    @renderSelectors('#product-'+productId)
+
+  onClickCancelProductInfoEditButton: (e) ->
+    productId=@$(e.target).data('product-id')
+    @productTableState[productId] = 'viewMode'
+    @renderSelectors('#product-'+productId)
+
+  onClickSaveProductInfo: (e) ->
+    productId= '' + @$(e.target).data('product-id') # make sure it is string
+    productStartDate= @$el.find('#product-startDate-'+productId).val()
+    productEndDate= @$el.find('#product-endDate-'+productId).val()
```

Comment:
Let's keep spaces between `=` and `+` operators, here and above
```suggestion
    productStartDate = @$el.find('#product-startDate-'+productId).val()
    productEndDate = @$el.find('#product-endDate-'+productId).val()
```

Reply:
okay. seems my config doesn't work for coffee files :joy: